bluemira/builders/EUDEMO/first_wall/divertor.py

```diff
@@ -0,0 +1,329 @@
+            LegPosition.OUTER: self.params["div_L2D_ob"],
+        }
+        self.separatrix_legs = get_separatrix_legs(self.equilibrium)
+        _, self.x_points = self.equilibrium.get_OX_points()
+
+    def reinitialise(self, params, **kwargs) -> None:
+        """
+        Initialise the state of this builder ready for a new run.
+        """
+        return super().reinitialise(params, **kwargs)
+
+    def mock(self):
+        """
+        Mock a solution to the builder's design problem.
+        """
+        pass
+
+    def run(self):
+        """
+        Run the builder's design problem.
+        """
+        pass
+
+    def build(self, **kwargs) -> Component:
+        """
+        Build the divertor component.
+        """
+        component = super().build(**kwargs)
+        component.add_child(self.build_xz())
+        return component
+
+    def build_xz(self) -> Component:
+        """
+        Build the divertor's components in the xz-plane.
+        """
+        component = Component("xz")
+
+        # Build the targets for each separatrix leg
+        inner_target = self.make_target(LegPosition.INNER, self.COMPONENT_INNER_TARGET)
+        outer_target = self.make_target(LegPosition.OUTER, self.COMPONENT_OUTER_TARGET)
+        for target in [inner_target, outer_target]:
+            component.add_child(target)
+
+        # Build the dome based on target positions
+        inner_target_end = self._get_wire_end_with_smallest(inner_target.shape, "z")
+        outer_target_start = self._get_wire_end_with_smallest(outer_target.shape, "z")
+        dome = self.make_dome(
+            inner_target_end, outer_target_start, label=self.COMPONENT_DOME
+        )
+        component.add_child(dome)
+
+        # Build the baffles
+        component.add_child(self._make_inner_baffle(inner_target.shape))
+        component.add_child(self._make_outer_baffle(outer_target.shape))
+
+        return component
+
+    def make_target(self, leg: LegPosition, label: str) -> PhysicalComponent:
+        """
+        Make a divertor target for a the given leg.
+        """
+        sols = self._get_sols_for_leg(leg)
+        leg_length = self._get_length_for_leg(leg)
+        # Just use the first scrape-off layer for now
+        point = find_point_along_wire_at_length(sols[0], leg_length)
+
+        # Create some vertical targets for now. Eventually the target
+        # angle will be derived from the grazing-angle parameter
+        target_length = self.params.div_Ltarg
+        target_coords = np.array(
+            [
+                [point[0], point[0]],
+                [point[1], point[1]],
+                [point[2] - target_length / 2, point[2] + target_length / 2],
+            ]
+        )
+        return PhysicalComponent(label, make_polygon(target_coords))
+
+    def make_dome(self, start: Sequence[float], end: Sequence[float], label: str):
+        """
+        Make a dome between the two given points
```

Comment:
Add a bit of a description here that the dome follows the flux lines from the inner target end points.